internal/model/json/validation/typedefs/fields.go

```diff
@@ -16,6 +16,8 @@ const (
 	maxPathLength     = 1024
 	hostnamePattern   = "[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?(.[a-zA-Z0-9]([-a-zA-Z0-9]*[a-zA-Z0-9])?)*$"
 
```

Comment:
Should we write a unit test that creates each resource & asserts the non-zero values that were set to the relevant JSON schema? This way, we could have the test fail in the event a default value is missing on the JSON schema.

Reply:
Right now we already have something like this checking if a resource with minimal configuration matches the default value: https://github.com/Kong/koko/blob/main/internal/resource/route_test.go#L38-L47

Do you think that's sufficient?

Reply:
I was actually talking about asserting that we're not missing any defaults on the JSON schema definition. What you're speaking of is a bit different & is simply just testing the defaults from the Kong Lua code are applied, not that we're setting the defaults on the JSON schema properly.